agent/internal/actor/treev4.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Sqreen. All Rights Reserved.
+// Please refer to our terms for more information:
+// https://www.sqreen.io/terms.html
+
+// Radix tree
+//
+// A radix-tree is used to efficiently store security actions by IP addresses
+// and networks.
+package actor
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/kentik/patricia"
+	"github.com/kentik/patricia/uint8_tree"
+)
+
+// IPv4 data-structure mapping CIDR IPv4 addresses to actions. The underlysing
+// radix-tree is used as an index to an array of actions. The number of actions
+// is limited.
+// Methods are not thread-safe.
```

Comment:
What happens if you run an action_reload then ?

Reply:
1. You fetch the list of actions.
2. You create a new store.
3. You lock the access to the store pointer.
4. You replace the current store with the new one.
5. You unlock the access to the store pointer.

Done [here](https://github.com/sqreen/go-agent/pull/46/files#diff-26ce5caf8aa8a08aeddacfc414185673R97).